### PR TITLE
Convert some more legacy caches to standardise cache container

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This converts some more legacy caches to Civi::cache containers

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

ping @totten @eileenmcnaughton

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/ACL/BAO/ACL.php [hunk 1]
```diff
@@ -688,7 +688,8 @@ public static function retrieve(&$params, &$defaults) {
    */
   public static function setIsActive($id, $is_active) {
     // note this also resets any ACL cache
-    CRM_Core_BAO_Cache::deleteGroup('contact fields');
+    $cache = Civi::cache('fields');
```
Comment (monishdeb) [inline CRM/ACL/BAO/ACL.php]: or simply 
```patch
-  CRM_Core_BAO_Cache::deleteGroup('contact fields');
+  Civi::cache('fields')->flush();
```

File: CRM/ACL/Form/ACL.php [hunk 2]
```diff
@@ -254,7 +254,8 @@ public static function formRule($params) {
    */
   public function postProcess() {
     // note this also resets any ACL cache
-    CRM_Core_BAO_Cache::deleteGroup('contact fields');
+    $cache = Civi::cache('fields');
+    $cache->flush();
```
Comment (monishdeb) [inline CRM/ACL/Form/ACL.php]: same as above

File: CRM/Admin/Form/Setting/Localization.php [hunk 3]
```diff
@@ -184,10 +184,12 @@ public function postProcess() {
...
     //cache contact fields retaining localized titles
     //though we changed localization, so reseting cache.
-    CRM_Core_BAO_Cache::deleteGroup('contact fields');
+    $contactCache = Civi::cache('fields');
+    $contactCache->flush();
 
     //CRM-8559, cache navigation do not respect locale if it is changed, so reseting cache.
-    CRM_Core_BAO_Cache::deleteGroup('navigation');
+    $navigationCache = Civi::cache('navigation');
+    $navigationCache->flush();
```
Comment (monishdeb) [inline CRM/Admin/Form/Setting/Localization.php]: ..same

File: CRM/Contact/BAO/Contact.php [hunk 4]
```diff
@@ -1321,9 +1321,10 @@ public static function importableFields(
 
     $fields = CRM_Utils_Array::value($cacheKeyString, self::$_importableFields);
```
Comment (monishdeb) [inline CRM/Contact/BAO/Contact.php]: or 

```php
$fields = CRM_Utils_Array::value($cacheKeyString, self::$_importableFields) ?: Civi::cache('fields')->get($cacheKeyString);
```

File: CRM/Core/BAO/CustomGroup.php [hunk 5]
```diff
@@ -259,7 +259,8 @@ public static function retrieve(&$params, &$defaults) {
    */
   public static function setIsActive($id, $is_active) {
     // reset the cache
-    CRM_Core_BAO_Cache::deleteGroup('contact fields');
+    $cache = Civi::cache('fields');
+    $cache->flush();
```
Comment (monishdeb) [inline CRM/Core/BAO/CustomGroup.php]: same as above

**Comment by seamuslee001** [comment]
thanks @monishdeb i have updated now to hopefully be more streamlined in the code

**Comment by monishdeb** [comment]
Nice, from code/style POV I am happy with the patch. Waiting for @totten @eileenmcnaughton for their feedback :)

File: CRM/Campaign/Selector/Search.php [hunk 6]
```diff
@@ -296,7 +296,7 @@ public function buildPrevNextCache($sort) {
 
       if (Civi::service('prevnext') instanceof CRM_Core_PrevNextCache_Sql) {
         // SQL-backed prevnext cache uses an extra record for pruning the cache.
-        CRM_Core_BAO_Cache::setItem($cacheKey, 'CiviCRM Search PrevNextCache', $cacheKey);
+        Civi::cache('prevNextCache')->set($cacheKey, $cacheKey);
```
Comment (eileenmcnaughton) [inline CRM/Campaign/Selector/Search.php]: @totten agreed that the effect of 
```
Civi::cache('prevNextCache')->set($cacheKey, $cacheKey);
```
is a clumsy way of saying
```

Civi::cache('prevNextCache')->set($cacheKey,  TRUE);
```

Comment (totten) [reply CRM/Campaign/Selector/Search.php]: Just to be clear - I support setting the value in the same way it was set before. But Eileen had a question about whether it was meaningful to do `set($cacheKey,$cacheKey)`, and (conceptually) these could be the same (although I'd make no assumption that they're actually equivalent in practice).

Also, note that `CRM_Core_BAO_PrevNextCache::cleanupCache()` specifically relies on this cache being stored in SQL. I'd recommend skeptical eye about using a cache-backend that is dynamically-chosen (as in `prevNextCache' => 'CiviCRM Search PrevNextCache' ... 'type' => ['*memory*', 'SqlGroup', 'ArrayCache']` in `Container.php` below) when there are hard-coded bits that assume this cache is SQL-backed...

Comment (seamuslee001) [reply CRM/Campaign/Selector/Search.php]: @totten good point have just pushed up change to ensure prevNextCache is sql Backed

Comment (seamuslee001) [reply CRM/Campaign/Selector/Search.php]: @totten should `CRM_Core_BAO_PrevNextCache:::cleanupCache()` be aware of the backed provider of prev next service?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 needs a rebase

**Comment by seamuslee001** [comment]
@eileenmcnaughton rebased now

**Comment by eileenmcnaughton** [comment]
@totten still hoping you will review this

**Comment by seamuslee001** [comment]
following a chat with @totten i have now split this PR up into separate ones for each cache group

[#14580 PrevNextCache](https://github.com/civicrm/civicrm-core/pull/14580)
[#14581 Navigation](https://github.com/civicrm/civicrm-core/pull/14581)
[#14582 Custom Data](https://github.com/civicrm/civicrm-core/pull/14582)
[#14583 contact fields](https://github.com/civicrm/civicrm-core/pull/14583)
[#14584 contact groups](https://github.com/civicrm/civicrm-core/pull/14584)

I have left some notes on what reviewers should be looking at when reviewing. Hopefully split up they might be easier to review

ping @monishdeb @mattwire

**Comment by seamuslee001** [comment]
closing this as it has been addressed via other PRs

**Pull request closed without merging**